lib/simplecov/merge_helpers.rb

```diff
@@ -4,13 +4,12 @@ module ArrayMergeHelper
     def merge_resultset(array)
       new_array = dup
       array.each_with_index do |element, i|
-        if element.nil? && new_array[i].nil?
-          new_array[i] = nil
-        else
-          local_value = element || 0
-          other_value = new_array[i] || 0
-          new_array[i] = local_value + other_value
-        end
+        pair = [element, new_array[i]]
+        new_array[i] = if pair.any?(&:nil?) && pair.map(&:to_i).all?(&:zero?)
```

Comment:
so, the result is `nil` if either set contains nil AND both are each one of either nil, zero, or some non-integer?